### DISCUSSION TITLE
How to add universal/app links to our applications?

### DISCUSSION THREAD

**Discussion by trydalch** [unoplatform/uno#19117] [Q&A] [answered]
With MAUI it's possible to set up [iOS Universal LInks](https://learn.microsoft.com/en-us/dotnet/maui/macios/universal-links?view=net-maui-8.0) and [Android app links](https://learn.microsoft.com/en-us/dotnet/maui/android/app-links?view=net-maui-8.0), but it's not clear to me how to make the necessary changes to the app delegate in Uno to support this.

Is this possible? And if so, how?

**Comment by MartinZikmund** [comment, marked as answer]
It should be possible to mostly follow this documentation with some adjustments - Uno Platform apps on mobile are still .NET Android/.NET iOS apps. Created https://github.com/unoplatform/uno/issues/19144 to add a tutorial to our docs

**Reply by trydalch** [reply]
Awesome.

A related question is how to know whether the app was launched via a link vs just opened from the home screen on iOS and Android? 

Should we be able to use the Launched Event Args there? Do we need to override something in the AppDelegate? 

We want to be able to launch our app via QR code and pass data through the code so we can navigate to a screen showing dynamic content.

**Reply by MartinZikmund** [reply]
As long as I am not mistaken, the activation will go through a different path in these two cases. Until support for this is implemented directly in Uno Platform, it will probably be easiest to use the `AppDelegate` overrides directly - following the Apple/Microsoft docs.
